src/biokbase/narrative/system.py

```diff
@@ -0,0 +1,64 @@
+import os
+import biokbase.auth
+import biokbase.narrative.clients as clients
+import time
+
+
+def strict_system_variable(var):
+    """
+    Returns a system variable.
+    If that variable isn't defined, or anything else happens, raises an exception.
+    """
+    result = system_variable(var)
+    if result is None:
+        raise ValueError(f'Unable to retrieve system variable: "{var}"')
+    return result
+
+
+def system_variable(var):
+    """
+    Returns a KBase system variable. Just a little wrapper.
+
+    Parameters
+    ----------
+    var: string, one of "workspace", "workspace_id", "token", "user_id"
+        workspace - returns the KBase workspace name
+        workspace_id - returns the numerical id of the current workspace
+        token - returns the current user's token credential
+        user_id - returns the current user's id
+
+    if anything is not found, returns None
+    """
+    var = var.lower()
+    if var == "workspace":
+        return os.environ.get("KB_WORKSPACE_ID", None)
+    elif var == "workspace_id":
+        ws_name = os.environ.get("KB_WORKSPACE_ID", None)
```

Comment:
Doesn't `os.environ.get("blahblahblah")` return None by default if it can't find the value?

Reply:
Yup. I really didn't modify that code, just copy-pasted. I'll make it better.

Reply:
Fixed up. Also swapped `BaseException` over to `Exception` for those try/catch blocks. That #TODO in there still holds - we should do some better error trapping there.